### PR TITLE
Adds PlatformDefaultTriggerType for FileTriggerType 

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>

Adds `PlatformDefaultTriggerType ` which is used to create file events. The main motivation for using this is the following:

1. Easier to swap out between `FileTriggerType`. This is useful for UNIX devs to debug issues that might affect Windows.
2. Removes various `#ifdef`s around the codebase.

With this change, `proxy_protocol_test` now passes on Windows. 

Additional Description:
Risk Level: Low (No-Op, Most of the changes are calculated at compile time
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by davinci26** [comment]
@envoyproxy/windows-dev

File: source/common/network/connection_impl.cc [hunk 1]
```diff
@@ -66,11 +66,8 @@ ConnectionImpl::ConnectionImpl(Event::Dispatcher& dispatcher, ConnectionSocketPt
   }
 
   // Libevent only supports Level trigger on Windows.
```
Comment (nigriMSFT) [inline source/common/network/connection_impl.cc:68]: This comment should probably be moved to file_event.h

Comment (davinci26) [reply source/common/network/connection_impl.cc]: makes sense. I moved it in 1d06ba3e527004c6e03757a3ea37314d9ac8b754

File: source/common/event/libevent_scheduler.h [hunk 2]
```diff
@@ -108,6 +108,17 @@ class LibeventScheduler : public Scheduler, public CallbackScheduler {
   static void onPrepareForStats(evwatch*, const evwatch_prepare_cb_info* info, void* arg);
   static void onCheckForStats(evwatch*, const evwatch_check_cb_info*, void* arg);
 
+  static constexpr int flagsBasedOnEventType() {
+    if (Event::PlatformDefaultTriggerType == FileTriggerType::Level) {
```
Comment (lizan) [inline source/common/event/libevent_scheduler.h:112]: Will this compile with `if constexpr`?

Comment (davinci26) [reply source/common/event/libevent_scheduler.h]: Yeah it compiles on msvc so I changed into it

**Pull request merged**